src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttributeList.cs

```diff
@@ -25,6 +25,20 @@ public TagHelperAttributeList()
         /// </summary>
         /// <param name="attributes">The collection to wrap.</param>
         public TagHelperAttributeList(IEnumerable<TagHelperAttribute> attributes)
+            : base (new List<TagHelperAttribute>(attributes))
```

Comment:
Tests are fine but do the MVC tag helpers require the allocations here? If yes, leave as-is. If no, make it `internal` (which shouldn't impact Razor tests).

Put another way, could the `<script>` and `<link>` tag helpers do something less obviously expensive e.g. reuse one attribute list? Or could we support the scenario with a copy constructor -- allowing the new `List<T>` to be allocated w/ a known size?